apps/storefront/src/pages/UserManagement/index.test.tsx

```diff
@@ -196,6 +199,132 @@ it('displays users with custom roles', async () => {
+    expect(troyMcClureHeading).not.toBeInTheDocument();
+
+    // once queries/mutations are changed to use real graphql variables,
+    // we can spy on the request "variables" instead of this hacky string matching
+    expect(getUsersQuerySpy).toHaveBeenCalledWith(expect.stringContaining('offset: 12'));
+  });
+});
+
+describe('when there is a previous page of results, the user click "Go to previous page" and the results load successfully', () => {
+  it('displays the previous page of results', async () => {
+    const firstElevenUsers = bulk(buildUserEdgeWith, 'WHATEVER_VALUES').times(11);
+    const troyMcClure = buildUserEdgeWith({ node: { firstName: 'Troy', lastName: 'McClure' } });
+    const usersPageOne = buildUsersResponseWith({
+      data: {
+        users: {
+          totalCount: 13,
+          pageInfo: { hasNextPage: true, hasPreviousPage: false },
+          edges: [...firstElevenUsers, troyMcClure],
+        },
+      },
+    });
+
+    const sallyCinnamon = buildUserEdgeWith({ node: { firstName: 'Sally', lastName: 'Cinnamon' } });
+    const usersPageTwo = buildUsersResponseWith({
+      data: {
+        users: {
+          totalCount: 13,
+          pageInfo: { hasNextPage: false, hasPreviousPage: true },
+          edges: [sallyCinnamon],
+        },
+      },
+    });
+
+    const getUsersResponse = vi
+      .fn<unknown[], UsersResponse>()
+      .mockReturnValue(buildUsersResponseWith(usersPageOne));
+
+    const getUsersQuerySpy = vi.fn();
+
+    server.use(
+      graphql.query('GetUserExtraFields', () =>
+        HttpResponse.json(buildUserExtraFieldsResponseWith('WHATEVER_VALUES')),
+      ),
+      graphql.query('GetUsers', ({ query }) => {
+        getUsersQuerySpy(query);
+
+        return HttpResponse.json(getUsersResponse());
+      }),
+    );
+
+    renderWithProviders(<UserManagement />, { preloadedState });
+
+    getUsersResponse.mockReturnValue(buildUsersResponseWith(usersPageTwo));
+
+    await userEvent.click(await screen.findByRole('button', { name: 'Go to next page' }));
+
+    const sallyCinnamonHeading = await screen.findByRole('heading', { name: 'Sally Cinnamon' });
+
+    expect(sallyCinnamonHeading).toBeInTheDocument();
+
+    getUsersResponse.mockReturnValue(buildUsersResponseWith(usersPageOne));
+
+    await userEvent.click(screen.getByRole('button', { name: 'Go to previous page' }));
+
+    expect(await screen.findByRole('heading', { name: 'Troy McClure' })).toBeInTheDocument();
+    expect(sallyCinnamonHeading).not.toBeInTheDocument();
+
+    // once queries/mutations are changed to use real graphql variables,
+    // we can spy on the request "variables" instead of this hacky string matching
+    expect(getUsersQuerySpy).toHaveBeenCalledWith(expect.stringContaining('offset: 0'));
+  });
+});
+
 describe('when the user clicks on the "edit" button for a user', () => {
   it('opens the "Edit User" modal', async () => {
```

Comment:
feel a bit awkward about this sort of describes, they look overly specific to the point there will mostly only be one `it` per block

Reply:
I normally aim for

```tsx
describe('the context in which the assertion applies', () => {
    it('defends this business assertion which must hold true', () => {
```

(using no `describe` when there are no pre-conditions or specific contexts required by the assertion)

Meaning they don't always end up being parents of multiple `it`s.

Would you prefer we collapsed them into something like the following?

```tsx
it('opens the "Edit User" modal when the user clicks on the "edit" button for a user', () => {
```

Reply:
I don't mind keeping it the way it is, just not used to it

Reply:
let's go with it as is, if it gets in the way or doesn't feel clear enough in test results we can change it up